### ISSUE TITLE
Geocaching.com: CAPTCHA problem seems to get bigger and bigger

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#16013]
On support, we have multiple people each day with the annoying CAPTCHA-problem when trying to use geocaching.com. It seems that the number of users unable to resolve this with our known standard methods (see below) is increasing.

I open this issue for discussion:
* Are there any ideas on how to ease the pain on this one? (is it maybe somehow possible to build a workaround such that users can log into geocaching.com with their web browser, then copy out some cookies and paste it into c:geo? Just blind-shooting ideas...)
* Are there any other tips to overcome the issue which are NOT yet part of the text below?

Thanks for feedback!

Current support text for CAPTCHA problem:
your login fails in c:geo as the geocaching website for security reasons requires you to solve a [CAPTCHA](https://en.wikipedia.org/wiki/CAPTCHA) before login is possible again. We cannot influence on c:geo side under which conditions Groundspeak decides to trigger such a CAPTCHA, however we have collected much data on how to prevent it.

As far as we know the CAPTCHA can be triggered by the following situations:

You several times tried to login with a wrong password
You tried to login from a suspicious IP address (e.g. a public WiFi, from a foreign country, using VPN or anonymizer services)
The mail address stored with your profile at geocaching.com is not verified or no longer valid
To overcome this problem please proceed as follows:

Use your phone browser to go to geocaching.com, login onto the website and once solve the shown CAPTCHA.
Afterwards go to c:geo Start - Menu - Settings - Services - geocaching.com and use "Update authorization" to doublecheck your username and password and try to login again
If the problem still remains, please check that the following before trying again as described above:

Check on the Geocaching website [settings page](https://www.geocaching.com/account/settings/emailpreferences), that your mail address is valid and verified
Make sure you are not using any VPN services (such as NordVPN, TOR, etc.) on your device
Try to use another internet connection (e.g. mobile network instead of Wifi)
Another trick that has been known to help is triggering a "captcha removal" via the official geocaching.com app (which you can find in Google Play Store):

If you have already installed the official app: try to log into this app. After that, update your authorization in c:geo (switch to c:geo, go to settings => services => geocaching.com and long-tap on "update or remove authorization", then re-enter your credentials)
If you have not yet installed the official app: install it, log into itm then update your authorization in c:geo (switch to c:geo, go to settings => services => geocaching.com and long-tap on "update or remove authorization", then re-enter your credentials). If this way is successful, you may uninstall the official app again at your own preference
Further help on login problems can also be found in our [user guide](https://manual.cgeo.org/en/loginproblems). There is a special section dedicated to the captcha problem further down this page.

If nothing helps, feel free to contact us again.

**Comment by eddiemuc** [comment]
the sentence "If nothing helps, feel free to contact us again." feels wrong btw, because if users really do contact us again I have no clue how to help them further :-)

**Comment by eddiemuc** [comment]
Users report on support that they are unable to use manual login because the keyboard does not open.

https://github.com/cgeo/cgeo/pull/16016

**Comment by fm-sys** [comment]
Just reproduced on my main device by setting a wrong password in cgeo. Can confirm that I'm now unable to log in anymore 😅

Adding 
```
android:focusable="true"
android:focusableInTouchMode="true"
```
to the webview may help according to some posts.

**Comment by eddiemuc** [comment]
> Adding 
> ```
> android:focusable="true"
> android:focusableInTouchMode="true"
> ```
> to the webview may help according to some posts.

Sadly it's not enough. Android WebView is really bitchy it seems...

I played around a lot with solutions suggested on the web. PR #16039 tries to fix the problem. A custom subclass of WebView is needed though.

**Comment by moving-bits** [comment]
After the recent fix published with 2024.09.09: Can this issue be closed?

**Comment by eddiemuc** [comment]
Jupp
